bundle/src/test/java/com/adobe/acs/commons/packagegarbagecollector/PackageGarbageCollectionJobTest.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.jackrabbit.vault.packaging.JcrPackage;
+import org.apache.jackrabbit.vault.packaging.JcrPackageManager;
+import org.apache.jackrabbit.vault.packaging.Packaging;
+import org.apache.jackrabbit.vault.packaging.VaultPackage;
+import org.apache.sling.event.jobs.Job;
+import org.apache.sling.event.jobs.consumer.JobConsumer;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+import static com.adobe.acs.commons.packagegarbagecollector.PackageGarbageCollectionScheduler.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PackageGarbageCollectionJobTest {
+
+    @Rule
+    public AemContext context = new AemContext(ResourceResolverType.JCR_MOCK);
+
+    PackageGarbageCollectionJob consumer;
+
+    @Mock
+    Packaging packaging;
+
+    @Before
+    public void setup() {
+        context.registerService(Packaging.class, packaging);
+        consumer = context.registerInjectActivateService(PackageGarbageCollectionJob.class);
+    }
+
+    @Test
+    public void testPackageJustOldEnough() {
+        mockPackaging(60, false);
+        Job job = mockJob();
+        assertEquals(JobConsumer.JobResult.OK, consumer.process(job));
+    }
+
+    @Test
+    public void testPackageOldEnough() {
+        mockPackaging(70, false);
+        Job job = mockJob();
+        assertEquals(JobConsumer.JobResult.OK, consumer.process(job));
+    }
+
+    @Test
+    public void testPackageTooYoung() {
+        mockPackaging(10, false);
+        Job job = mockJob();
+        assertEquals(JobConsumer.JobResult.OK, consumer.process(job));
+    }
+
+    @Test
+    public void testFailingToRemovePackages() {
+        mockPackaging(61, true);
+        Job job = mockJob();
+        assertEquals(JobConsumer.JobResult.FAILED, consumer.process(job));
+    }
+
+    void mockPackaging(Integer daysAgo, boolean fail) {
+        try {
+            JcrPackageManager packageManager = mock(JcrPackageManager.class);
+            when(packaging.getPackageManager(any())).thenReturn(packageManager);
+            JcrPackage mockPackage = mockPackage(daysAgo);
+            when(packageManager.listPackages(anyString(), anyBoolean())).thenReturn(Collections.singletonList(mockPackage));
+            if (fail) {
+                doThrow(new RepositoryException()).when(packageManager).remove(any());
+            }
+        } catch (RepositoryException | IOException e) {
```

Comment:
no need to do that, just throw the exception; any non-caught exception makes the test fail

Reply:
Removed